### PR TITLE
Add internal printchplenv variable CHPL_LLVM_TARGET_CPU

### PR DESCRIPTION
**Pull request by mppf**
Follow-up to PR #21944.

This PR is intended to fix testing failures with `test/runtime/configMatters/mem/fragmentation/crossThreadFragmentation.chpl` (as well as other tests that use `--fast`) on Cray systems when not using the LLVM backend. 

The new extern blocks support relies on a refactor of some of the code computing clang args. The clang args are now computed even if `CHPL_TARGET_COMPILER!=llvm` in order to support extern blocks. Computing the clang args includes some additional checking that the target architecture is valid for clang. We were running into problems with that where the architecture name in `CHPL_TARGET_CPU` / (from the loaded Cray prgenv module in the problematic configuration) did not match the clang name. However, we already have the ability to translate such a CPU name to a name used by clang. This PR adjusts things to use the translated name in this case.

To do so, this PR adds a new internal printchplenv variable `CHPL_LLVM_TARGET_CPU` which is similar to `CHPL_TARGET_BACKEND_CPU` but always uses the target CPU translation table in order to always use the LLVM/clang CPU name, even when `CHPL_TARGET_COMPILER!=llvm`, to support extern blocks. Note that `CHPL_TARGET_BACKEND_CPU` is still used by Makefiles (as `CHPL_MAKE_TARGET_BACKEND_CPU`), but it is no longer used within the compiler.

Reviewed by @riftEmber - thanks!

- [x] full local testing

### CODE REVIEW

File: util/chplenv/printchplenv.py [hunk 1]
```diff
@@ -127,6 +126,7 @@
     ChapelEnv('  CHPL_LLVM_CLANG_C', INTERNAL),
     ChapelEnv('  CHPL_LLVM_CLANG_CXX', INTERNAL),
     ChapelEnv('  CHPL_LLVM_STATIC_DYNAMIC', INTERNAL),
+    ChapelEnv('  CHPL_LLVM_TARGET_CPU', INTERNAL),
```
Comment (riftEmber) [inline util/chplenv/printchplenv.py:129]: What do the spaces at the beginning mean? I see they weren't there for the previous `CHPL_TARGET_BACKEND_CPU` variable

File: util/chplenv/chpl_cpu.py [hunk 2]
```diff
@@ -193,6 +193,17 @@ def get(flag, map_to_compiler=False, get_lcd=False):
 
     return cpu_tuple(argname or 'none', cpu or 'unknown')
 
+@memoize
+def get_llvm_target_cpu():
```
Comment (riftEmber) [inline util/chplenv/chpl_cpu.py:197]: It looks to me like the `get` function with `map_to_compiler=True`, when `compiler_val` is one of `'llvm', 'gnu', 'clang'`, is doing the same `cpu_llvm_synonyms` lookup as the new `get_llvm_target_cpu` function does regardless of compiler. I get why the new function would be helpful in the case of a compiler besides one of those, but I see that one of the failing configurations was with `gnu` compiler, failing with the same issue. So I think I must be missing something -- why would we be seeing this failure with the `gnu` compiler as well?

**Review by riftEmber — COMMENTED** [review]
This is an unfamiliar code area to me, but I think I understand the problem well enough to review it -- just have some (maybe basic) questions.

File: util/chplenv/printchplenv.py (hunk 1 above)
Comment (mppf) [reply util/chplenv/printchplenv.py:129]: They just make it print out with spaces to seem nested (try `util/printchplenv --all` to see what I mean)

File: util/chplenv/chpl_cpu.py (hunk 2 above)
Comment (mppf) [reply util/chplenv/chpl_cpu.py:197]: We compute the flags for running `clang` in order to handle extern blocks & do so even if we are generating C for another compiler. That was the original issue.

**Review by riftEmber — APPROVED** [review]

**Pull request merged**